contracts/03-tamagotchi-nft/src/lib.rs

```diff
@@ -129,6 +131,32 @@ extern fn handle() {
                     msg::reply(TmgEvent::Slept, 1).expect("Error in a reply'tamagotchi::slept'");
                 }
             }
+            TmgAction::Transfer(account) => {
+                tmg.owner = account;
```

Comment:
One should reset the `approved_account` after transferring the ownership.

Reply:
<img width="703" alt="image" src="https://github.com/Jackliu-miaozi/Jackliu-gear-academy/assets/71253778/90b4f981-468d-448e-a95c-037b5168e5e8">